apis/nucleus/src/components/listbox/interactions/listbox-keyboard-navigation.js

```diff
@@ -99,38 +132,37 @@ export function getFieldKeyboardNavigation({
 }
 
 export function getListboxInlineKeyboardNavigation({
-  setKeyboardActive,
+  keyboard,
   hovering,
   setHovering,
   updateKeyScroll,
   containerRef,
   currentScrollIndex,
-  app,
-  appSelections,
+  isModal,
   constraints,
 }) {
-  const focusInsideListbox = (element) => {
-    const fieldElement = element.querySelector('.search input, .value.selector, .value, .ActionsToolbar-item button');
-    setKeyboardActive(true);
-    if (fieldElement) {
-      fieldElement.focus();
-    }
-  };
-
-  const changeFocus = (event) => {
-    if (event.target?.classList.contains('listbox-container')) {
-      // Focus currently on a listbox
-      // Esc on a list box should move the focus to its parent, i.e. a filter pane if any
-      setKeyboardActive(false);
+  const blur = (event) => {
+    const { target } = event;
+    const isFocusedOnListbox = target.classList.contains('listbox-container');
+    if (isFocusedOnListbox) {
+      // Move the focus from listbox container to the viz container.
+      keyboard.blur(true);
     } else {
-      // Focus currently on a row
-      // Esc on a row should move the focus to its parent, i.e. a listbox
-      // First unfocus the row
-      event.target?.setAttribute('tabIndex', '-1');
-      event.target?.blur();
-      // Then focus the listbox
-      event.currentTarget?.setAttribute('tabIndex', '0');
-      event.currentTarget?.focus();
+      // Move focus from row to listbox container:
+
+      // 1. Remove last-focused class from row siblings.
+      const container = target.closest('.listbox-container');
+      container?.querySelectorAll('.last-focused').forEach((elm) => {
```

Comment:
how does last-focused work if you have multiple listboxes?
only used in one of them that was last focused or all of the could have one last-focused?

Reply:
All of them could have a 'last-focused' that remains also after the listbox has been escaped (not sure if it's supposed to work like this…). But focusing should only be performed from a scope of a listbox, which should rule out that it goes into another listbox e.g.